src/mlpack/methods/softmax_regression/softmax_regression_function.hpp

```diff
@@ -38,12 +37,25 @@ class SoftmaxRegressionFunction
   const arma::mat InitializeWeights();
 
   /**
+   * Initialize Softmax Regression weights(trainable parameters) with
+   * the given parameters.
+   * @param featureSize The features size of the training set
+   * @param numClasses Number of classes for classification.
+   * @param fitIntercept Intercept term flag.
+   * @return weights after initialize
+   */
+  static const arma::mat InitializeWeights(const size_t featureSize,
+                                           const size_t numClasses,
+                                           const bool fitIntercept = false);
```

Comment:
I think for now I'll go with `weights` as a parameter (in part for consistency with other code), but if you run some quick simulations to determine that gcc and clang are doing RVO correctly here, we can change it.